packages/ui-components/src/components/ComboBox/ComboBox.component.tsx

```diff
@@ -312,128 +335,134 @@ export const ComboBox: React.FC<ComboBoxProps> = ({
-                  <span
-                    className={`
-                        juno-combobox-icon-container 
-                        ${iconContainerStyles} 
-                        ${disabled ? "jn-opacity-50" : ""}
-                      `}
-                  >
-                    <Icon
-                      icon={isValid ? "checkCircle" : "dangerous"}
-                      color={isValid ? "jn-text-theme-success" : "jn-text-theme-error"}
-                    />
-                  </span>
-                ) : (
-                  ""
-                )}
-
-                {!hasError && !isLoading ? (
-                  <Combobox.Button
-                    className={`
-                        juno-combobox-toggle 
-                        ${buttonStyles}
-                        ${disabled ? disabledButtonStyles : ""}
-                        ${isInvalid ? "juno-combobox-toggle-invalid " + invalidButtonStyles : ""} 
-                        ${isValid ? "juno-combobox-toggle-valid " + validButtonStyles : ""}  
-                        ${isValid || isInvalid ? "" : defaultButtonStyles} 
-                      `}
-                  >
-                    {({ open }) => <Icon icon={open ? "expandLess" : "expandMore"} />}
-                  </Combobox.Button>
-                ) : null}
-              </div>
-            </Float.Reference>
-
-            {createPortal(
-              <Float.Content>
-                <Combobox.Options
-                  unmount={false}
-                  className={`
-                    juno-combobox-options 
-                    ${menuStyles}
-                  `}
-                >
+            {!hasError && !isLoading ? (
+              <ComboboxButton
+                className={`
+                  juno-combobox-toggle 
+                  ${buttonStyles}
+                  ${disabled ? disabledButtonStyles : ""}
+                  ${isInvalid ? "juno-combobox-toggle-invalid " + invalidButtonStyles : ""} 
+                  ${isValid ? "juno-combobox-toggle-valid " + validButtonStyles : ""}  
+                  ${isValid || isInvalid ? "" : defaultButtonStyles} 
+                `}
+              >
+                {({ open }) => {
+                  // Update our open state when Headless UI updates it
+                  useEffect(() => {
+                    if (open !== isOpen) {
+                      setIsOpen(open)
+                    }
+                  }, [open])
+
+                  return <Icon icon={open ? "expandLess" : "expandMore"} />
+                }}
+              </ComboboxButton>
+            ) : null}
+          </div>
+          {isOpen &&
+            createPortal(
+              <div
+                ref={refs.setFloating}
+                className={`juno-combobox-options ${menuStyles}`}
+                style={{
+                  position: strategy,
+                  top: y ?? 0,
+                  left: x ?? 0,
+                  zIndex: 999,
```

Comment:
`z-index` is a complicated topic, please don't mess with the z-index here in the tsx. We very carefully craft z-index of all things in css. I don't want to introduce another source of truth for the z-index.